docs/dev/cloudFoundry/CloudFoundryPlugin_Basics.md

```diff
@@ -0,0 +1,78 @@
+# Cloud Foundry Plugin
+
+## Basic information of Cloud Foundry
+Cloud Foundry is a open-source Platform as a Service Model (PaaS) which allows everyone to deploy easily an existing application on an infrastructure e.g. AWS, vSphere or OpenStack.
+Cloud Foundry installs everything what the application needs to compile and run like OS, buildpack with all languages, libraries, and services that the app uses.
+
+For further information you can have a look at the official [Cloud Foundry Documentation](https://docs.cloudfoundry.org/concepts/overview.html).
+## Transformation
+For the further document it is assumed that the CloudFoundry-Platform is already installed on the target infrastructure. If not the process of deployment throws an error. There is a well documented manual how to install Cloud Foundry on the [CF-Documentation](https://docs.cloudfoundry.org/deploying/index.html).
+### Workflow
+This workflow describes the doing of the plugin this means the system did some steps before.
+
+plugin...
+- gets data model of core
+- parse the nodes
+  - go through the given model
+  - check every node what type it is
+  - compare the type with a fix internal model
+    - this fix internal model represents supported nodetypes&connections
+    - for every nodetypes exist a rule what to do with CloudFoundry e.g. nodetype "mysql-Database" which is connected to the application-node. That means for CloudFoundry that a mysql-service have to be created in the deployment-script (before the deployment of the application). Also there has to be a entry in the manifest which binds the service to the application.
+- reads the information out of the nodes (e.g. applicationname, size of storage, ...) and writes it into the manifest
+  - for this the plugin has to know the different properties of the nodetypes
+  - for every property there is a rule what to do with
+- creates a deployment script. The script uses CF-CLI commands
+  - so the plugin has to know the order of creation.
+- return the created artifacts (manifest.yml and optional a deployment-script) respectively the path to it
+
+In summary the plugin needs a list of the supported nodes of the Tosca-Model and the properties of it. The plugin knows what to do with this information. Therefore it needs for every nodetype and every propertie a rule.
```

Comment:
What do you mean by "supported nodes"?

Reply:
i thought the plugin supports a few special nodetypes